app/main/errors.py

```diff
@@ -1,67 +1,21 @@
-from urllib.parse import quote_plus
-
-from flask import redirect, render_template, request
-
 from dmapiclient import APIError
 from dmcontent.content_loader import QuestionNotFoundError
 from dmutils.s3 import S3ResponseError
+from dmutils.errors import render_error_page
 
 from app.main import main
 
 
 @main.app_errorhandler(APIError)
 def api_error_handler(e):
-    return _render_error_page(e.status_code)
+    return render_error_page(status_code=e.status_code)
 
 
 @main.app_errorhandler(S3ResponseError)
 def s3_response_error_handler(e):
-    return _render_error_page(503)
+    return render_error_page(status_code=503)
 
 
 @main.app_errorhandler(QuestionNotFoundError)
 def content_loader_error_handler(e):
-    return _render_error_page(400)
-
-
-@main.app_errorhandler(401)
-def page_unauthorized(e):
-    if request.method == 'GET':
-        return redirect('/user/login?next={}'.format(quote_plus(request.path)))
-    else:
-        return redirect('/user/login')
-
-
-@main.app_errorhandler(400)
-def bad_request(e):
-    return _render_error_page(400, e.description or None)
-
-
-@main.app_errorhandler(404)
-def page_not_found(e):
-    return _render_error_page(404)
-
-
-@main.app_errorhandler(500)
```

Comment:
If we're removing these handlers, how does the app handle a 500 now?

Reply:
In `dmutils.flask_init`. We register handlers for a bunch of common status codes.

Reply:
ah, nice!